### PR TITLE
Avoid using Cursor.schema API when database is None

### PR DESCRIPTION
**Pull request by ueshin**
### Description

Avoids using `Cursor.schema` API when `database` is `None`.

`Cursor.schema` API tries to collect schema from all catalogs when `database` is `None`, but we should collect from only the current catalog.

### CODE REVIEW

**Comment by ueshin** [comment]
This is based on #98.

File: dbt/adapters/databricks/impl.py [hunk 1]
```diff
@@ -39,13 +39,15 @@ class DatabricksAdapter(SparkAdapter):
 
     def list_schemas(self, database: Optional[str]) -> List[str]:
         """Get a list of existing schemas in database."""
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/impl.py:41]: Let's also update this comment to specify the behavior when `database` is None?

File: dbt/adapters/databricks/impl.py [hunk 2]
```diff
@@ -39,13 +39,15 @@ class DatabricksAdapter(SparkAdapter):
...
-        results = self.connections.list_schemas(database=database)
+        if database is not None:
+            results = self.connections.list_schemas(database=database)
+        else:
+            results = self.execute_macro(LIST_SCHEMAS_MACRO_NAME, kwargs={"database": database})
         return [row[0] for row in results]
 
     def check_schema_exists(self, database: Optional[str], schema: str) -> bool:
         """Check if a schema exists."""
```
Comment (allisonwang-db) [inline dbt/adapters/databricks/impl.py:49]: ditto

**Review by allisonwang-db — APPROVED** [review]

File: dbt/adapters/databricks/impl.py (hunk 2 above)
Comment (ueshin) [reply dbt/adapters/databricks/impl.py:49]: Maybe we don't need here as it call `list_schemas` now?

Comment (allisonwang-db) [reply dbt/adapters/databricks/impl.py:54]: Sounds good.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**